main.go

```diff
@@ -26,84 +24,74 @@ var (
+			logrus.SetLevel(logrus.InfoLevel)
+		} else {
+			logrus.SetLevel(logLevel)
+		}
 	}
 
-	ceHandler, err := cloudevents.NewHTTPReceiveHandler(ctx, p, gotEvent)
-	if err != nil {
-		log.Fatalf("failed to create handler: %s", err.Error())
-	}
+	log.Printf("Starting %s", serviceName)
 
 	http.HandleFunc("/", HTTPGetHandler)
-	http.Handle(env.Path, ceHandler)
-	err = http.ListenAndServe(":8080", nil)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	return 0
-}
+	go func() {
+		err := http.ListenAndServe(":8080", nil)
+		if err != nil {
+			log.Fatalf("Error with HTTP server: %e", err)
+		}
+	}()
 
-// gotEvent processes an incoming CloudEvent
-func gotEvent(event cloudevents.Event) error {
-	keptnContext, err := event.Context.GetExtension("shkeptncontext")
+	clusterConfig, err := rest.InClusterConfig()
 	if err != nil {
-		return fmt.Errorf("cloud event does not contain the field 'shkeptncontext'")
+		log.Fatalf("unable to create kubernetes cluster config: %e", err)
 	}
 
-	shkeptncontext, err := types.ToString(keptnContext)
+	kubeClient, err := kubernetes.NewForConfig(clusterConfig)
 	if err != nil {
-		return fmt.Errorf("field 'shkeptncontext' can not be parsed as keptn context")
-	}
-
-	logger := keptncommon.NewLogger(shkeptncontext, "", utils.ServiceName)
-
-	// convert v0.1.4 spec monitoring.configure CloudEvent into a v0.2.0 spec configure-monitoring.triggered CloudEvent
-	if event.Type() == keptn.ConfigureMonitoringEventType {
-		event.SetType(keptnv2.GetTriggeredEventType(keptnv2.ConfigureMonitoringTaskName))
+		log.Fatalf("unable to create kubernetes client: %e", err)
 	}
 
-	keptnHandler, err := keptnv2.NewKeptn(&event, keptncommon.KeptnOpts{})
+	log.Fatal(sdk.NewKeptn(
+		serviceName,
+		sdk.WithTaskHandler(
+			monitoringTriggeredEvent,
+			eventhandling.NewConfigureMonitoringEventHandler(),
+			prometheusTypeFilter),
+		sdk.WithTaskHandler(
+			getSliTriggeredEvent,
+			eventhandling.NewGetSliEventHandler(*kubeClient),
+			prometheusSLIProviderFilter),
+		sdk.WithLogger(logrus.New()),
+	).Start())
+}
 
-	if err != nil {
-		return fmt.Errorf("could not create Keptn handler: %v", err)
+func prometheusSLIProviderFilter(keptnHandle sdk.IKeptn, event sdk.KeptnEvent) bool {
+	data := &keptnv2.GetSLITriggeredEventData{}
+	if err := keptnv2.Decode(event.Data, data); err != nil {
+		keptnHandle.Logger().Errorf("Could not parse test.triggered event: %s", err.Error())
+		return false
 	}
 
-	clusterConfig, err := rest.InClusterConfig()
-	if err != nil {
-		// TODO: Send Error log event to Keptn
-		return fmt.Errorf("unable to create kubernetes cluster config: %w", err)
-	}
+	return data.GetSLI.SLIProvider == "prometheus"
+}
 
-	kubeClient, err := kubernetes.NewForConfig(clusterConfig)
-	if err != nil {
-		// TODO: Send Error log event to Keptn
-		return fmt.Errorf("unable to create kubernetes client: %w", err)
+func prometheusTypeFilter(keptnHandle sdk.IKeptn, event sdk.KeptnEvent) bool {
```

Comment:
please add a comment like "filtering configure-monitoring events for prometheus"